client/pages/viewcards.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+export default class ViewCards extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = ({ cards: [] });
+  }
+
+  render() {
+    fetch(`/api/cards/${this.props.deckId}`)
+      .then(res => res.json())
+      .then(data => {
+        this.setState({ cards: data });
+      })
+      .catch(err => console.error(err));
```

Comment:
We do not fetch in render. Render should only return react elements

Reply:
Thanks for the knowledge that the componentDidMount gets destroyed on the hashchange! Made this a much better solution

![image](https://user-images.githubusercontent.com/109567499/200924079-bc40779c-2bb5-4655-b9bc-fdcebbcde33b.png)